clayui.com/src/components/Editor/index.js

```diff
@@ -14,28 +15,56 @@ import theme from '../../utils/react-live-theme';
 
 const spritemap = '/images/icons/icons.svg';
 
+function formatCode(code) {
+	try {
+		return prettier.format(code, {
+			parser: 'babel',
+			plugins: [parserBabylon],
+		});
+	} catch (e) {
+		// eslint-disable-next-line
+		console.log(e);
+
+		return;
+	}
+}
+
 const Editor = ({
 	code,
 	disabled = false,
 	imports,
 	preview = true,
 	scope = {},
 }) => {
-	try {
-		code = prettier.format(code, {
-			parser: 'babel',
-			plugins: [parserBabylon],
+	let reactSnippet = {
+		code: '',
+		name: 'React',
+	};
+
+	if (Array.isArray(code)) {
+		const codeToFormat = [...code];
+
+		codeToFormat.forEach((snippet) => {
+			if (snippet.name !== 'JSP') {
+				snippet.code = formatCode(snippet.code);
+			}
 		});
-	} catch (e) {
-		// eslint-disable-next-line
-		console.log(e);
+
+		code = codeToFormat;
+
+		reactSnippet = code.find((snippet) => snippet.name === 'React');
```

Comment:
Its also odd that we are checking for the name of the code here specifically against react. I would expect this component to just use the first item in the array of code objects.

Reply:
Alright, that was my initial approach too, but then I assumed that checking for the name would be more robust so I changed it, in case someone comes in and doesn't know that the React snippet must be at index `0`. @matuzalemsteles got any thoughts on this one?

Reply:
Yeah, I would agree that the use of the name would be better here, just in case someone puts the JSP snippet at index `0` 🤷‍♂️, but just getting the index zero is simpler too but doing this I would create a constant with `const REACT = 0;` for when we look at this stretch in the future, it will be clearer.